parsec/core/cli/run.py

```diff
@@ -55,3 +57,121 @@ def run_mountpoint(config, device, mountpoint, timestamp, **kwargs):
         config = config.evolve(mountpoint_base_dir=Path(mountpoint))
     with cli_exception_handler(config.debug):
         trio_run(_run_mountpoint, config, device, timestamp)
+
+
+async def _import_file(workspace_fs, src, dest):
+    try:
+        await workspace_fs.touch(dest)
+    except FileExistsError:
+        await workspace_fs.truncate(dest, 0)
+    with open(src, "rb") as fd:
+        read_size = 0
+        while True:
+            chunk = fd.read(DEFAULT_BLOCK_SIZE)
+            if not chunk:
+                break
+            await workspace_fs.write_bytes(dest, chunk, read_size)
+            read_size += len(chunk)
+
+
+def _local_access_digests(workspace_fs, src):
+    digests = []
+    with open(src, "rb") as fd:
+        read_size = 0
+        while True:
+            chunk = fd.read(DEFAULT_BLOCK_SIZE)
+            if not chunk:
+                break
+            digests.append(HashDigest.from_data(chunk))
+            read_size += len(chunk)
+    return digests
+
+
+async def _update_file(workspace_fs, entry_id, path, absolute_path):
+    remote_file_manifest = await workspace_fs.remote_loader.load_manifest(entry_id)
+    local_access_digests = _local_access_digests(workspace_fs, path)
+    remote_access_digests = [access.digest for access in remote_file_manifest.blocks]
+    if local_access_digests != remote_access_digests:
+        await _import_file(workspace_fs, path, absolute_path)
+        print("update %s" % absolute_path)
+        await workspace_fs.sync()
```

Comment:
@vxgmichel is there a way to use the transaction to ask for sync of only a certain file/folder ? this would come handy here !

Reply:
Yes, `workspace_fs.sync_by_id(entry_id, remote_changed=False, recursive=False)`:
https://github.com/Scille/parsec-cloud/blob/f8f7f9e2dfcd26633772e53c181c05c60b683b06/parsec/core/fs/workspacefs/workspacefs.py#L576-L578